src/components/CurrentPlaylist.js

```diff
@@ -2,11 +2,17 @@ import React from 'react';
 import macaroon from '../assets/img/Macaroonicon.png';
 import { clonePlaylist, fetchSomeTracks } from '../helpers/spotifyHelpers';
 import TrackItem from './TrackItem';
+import { PlaylistHeader } from './PlaylistHeader';
+import EditPlaylistDetails from './EditPlaylistDetails';
+import { PlaylistDetails } from './PlaylistDetails';
+import { PlaylistButtons } from './PlaylistButtons';
 import { ErrorMessage } from './ErrorMessage';
 
 class CurrentPlaylist extends React.Component {
   state = {
-    errorMessage: '',
+    warning: '',
+    draftPlaylist: this.props.playlist,

```

Comment:
`draftPlaylist` added so we can edit playlist details without them being reflected in other areas of the App until we submit changes and get a response of OK from the server. And so the playlist can be reset if we cancel the edit or if the server request gets rejected.